data-import/desktop-csv-import/desktop-csv-import.adoc

```diff
@@ -0,0 +1,420 @@
+CREATE CONSTRAINT ON (o:Order) ASSERT o.id IS UNIQUE;
+----
+
+=== Cypher
+
+Now we are ready to create the Cypher for creating the data in the graph!
+We will use `CREATE` statements where we are sure that we will not have duplicate rows in our CSV file.
+For values we know exist in the database already, we will first use `MATCH` to find them before executing any updates or inserts.
+
+.Products
+To start, let us load the products into the graph.
+We start with our `LOAD CSV` statement from above, then we add the Cypher to create the data from the CSV into our graph model.
+We can use the `CREATE` statement, as we feel confident that we have our constraint in place, no duplicate rows in our file, and no existing data in our graph.
+The properties will be set to the converted values we handled earlier in this guide.
+
+[source, cypher]
+----
+LOAD CSV FROM 'file:///desktop-csv-import/products.csv' AS row
+CREATE (p:Product {productId: toInteger(row[0])})
+  SET p.productName = row[1], p.unitCost = toFloat(row[2])
+RETURN count(p)
+----
+
+If you run that statement, it will return the number of product nodes (`count(p)`) that were created in the database.
+You can cross-check that number with the number of rows in the CSV file from earlier (77 rows in products.csv).
+You can also run a validation query to return a sample of nodes and review that the properties look accurate.
+
+[source, cypher]
+----
+//validate products loaded correctly
+MATCH (p:Product)
+RETURN p LIMIT 20
+----
+
+image::{img}desktop_import_verify_products.jpg[link="{img}desktop_import_verify_products.jpg",role="popup-link"]
+
+.Orders
+Next, we can load the orders.
+Again, since we feel confident our constraint will catch violations and our CSV file data has no duplicates, we can use the `CREATE` statement to insert the data more efficiently.
+Just as with products, we start with the `LOAD CSV` command, then add Cypher statements and include our data conversions.
+
+[source, cypher]
+----
+LOAD CSV WITH HEADERS FROM 'file:///desktop-csv-import/orders.csv' AS row
+CREATE (o:Order {orderId: toInteger(row.orderID)})
+  SET o.orderDateTime = datetime(replace(row.orderDate,' ','T')), o.shipCountry = row.shipCountry
+RETURN count(o)
+----
+
+We can also run a validation query, as before, to verify the graph data looks correct.
+
+[source, cypher]
+----
+//validate orders loaded correctly
+MATCH (o:Order)
+RETURN o LIMIT 20
+----
+
+image::{img}desktop_import_verify_orders.jpg[link="{img}desktop_import_verify_orders.jpg",role="popup-link"]
+
+.Order-details
+Last, but not least, we create the relationship between the products and the orders.
+Since we expect all of our products and all of our orders to already exist in the graph (that data should have been loaded with the last two files), then we start with `MATCH` to find the existing `Product` and `Order` nodes.
+Then, since we know relationships do not yet exist in the graph, the `CREATE` statement will add the new relationship.
```

Comment:
For all rows where either of the two is missing, no relationship is created.

Reply:
Not sure you would want that if you were importing business data from another system. You wouldn't want it to arbitrarily create and Order or Product that wasn't in your relational table to begin with.....at least I wouldn't think so. Do you disagree?